torchtune/utils/_device.py

```diff
@@ -88,12 +110,15 @@ def _validate_device_from_env(device: torch.device) -> None:
     local_rank = _get_local_rank()
 
     # Check if the device index is correct
-    if device.type == "cuda" and local_rank is not None:
+    if (
+        device.type in DeviceSupport.get_cuda_like_device_types()
+        and local_rank is not None
+    ):
         # Ensure device index matches assigned index when distributed training
         if device.index != local_rank:
             raise RuntimeError(
                 f"You can't specify a device index when using distributed training. \
-                Device specified is {device} but was assigned cuda:{local_rank}"
+                Device specified is {device} but was assigned cuda device:{local_rank}"
```

Comment:
I think this log message was not super clear before but this is also unclear. Maybe something like 
```suggestion
                Device specified is {device} but local rank is {local_rank}"
```
(assuming that NPU devices also contain rank in their string representation?)

Btw a higher-level question here: it was mentioned in a previous comment that there were issues running some of the distributed training scripts on NPU. Did that all get sorted out?

Reply:
Yes, NPU devices also contain.
There are still some issues with adapting NPU for distributed scripts. I’m working on it and will include the updates in a future PR. Thanks for the reminder!

Reply:
In that case we probably don't even need to compare device index against device count for NPU, right? (At least not until those changes land.) Though I think it's fine to leave this as is since handling NPU separately here may be messier